x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/wizard.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component, Fragment } from 'react';
+import { EuiEmptyPrompt, EuiPanel } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { createNewIndexAndPattern } from './utils/indexing_service';
+import { getFileUpload } from '../../../kibana_services';
+import { RenderWizardArguments } from '../layer_wizard_registry';
+import { VectorLayer } from '../vector_layer';
+import { ESSearchSource } from '../../sources/es_search_source';
+import { ADD_LAYER_STEP_ID } from '../../../connected_components/add_layer_panel/view';
+import { IndexNameFormProps } from '../../../../../file_upload/public';
+
+export const ADD_VECTOR_DRAWING_LAYER = 'ADD_VECTOR_DRAWING_LAYER';
+
+interface State {
+  indexName: string;
+  indexError: string;
+  indexingTriggered: boolean;
+  indexPatternId: string;
+  indexNameFormComponent: React.ComponentType<IndexNameFormProps> | null;
+}
+
+export class NewVectorLayerEditor extends Component<RenderWizardArguments, State> {
+  state: State = {
+    indexName: '',
+    indexError: '',
+    indexingTriggered: false,
+    indexPatternId: '',
+    indexNameFormComponent: null,
+  };
+
+  componentDidMount() {
+    this._loadIndexNameFormComponent();
+  }
+
+  async _loadIndexNameFormComponent() {
+    this.setState({
+      indexNameFormComponent: await getFileUpload().getIndexNameFormComponent(),
```

Comment:
I didn't really love it either so agreed it's good to revise.

The plugin's `start` method is called before `mount` and seems like the appropriate place to init the components, so I've done it from there. I also updated the File Upload component to follow the same pattern since it had been doing the same thing via component state.

Reply:
discussed offline. Cannot be in start, because it cannot capture the async-ness.

In mount, it'd be something like:


```
async mount(params: AppMountParameters) { 
   const { renderApp } = await lazyLoadMapModules(); 
   kibana_services.setIndexNameFormComponent(await plugins.fileUpload.getIndexNameFormComponent()); //captures async. can get component synchornously with `kibana_services.getIndexNameFormComponent`.
   return renderApp(params); 
 }, 
```

@nreese suggestion is a little more sophisticated (https://github.com/elastic/kibana/pull/96913#discussion_r625329705) but OK to imho. Advantage is that it hides the asynciness completely for consumers.

Reply:
made these async in the file-upload plugin itself in https://github.com/elastic/kibana/pull/100278